README.md

```diff
@@ -115,6 +116,10 @@ By default, this action will try to authenticate Pulumi with the
 `PULUMI_ACCESS_TOKEN` then you will need to specify an alternative backend via
 the `cloud-url` argument.
 
+If you are using any Pulumi plugins then you should also set the `GITHUB_TOKEN`
+environment variable in order to avoid hitting GitHub rate limits. You can set
+it to the default `${{ secrets.GITHUB_TOKEN }}`.
```

Comment:
I guess users need to set a personal GitHub-token?

Reply:
No, the GITHUB_TOKEN secret is always added by GitHub automatically: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret

However, they would have the option of using another token if they wish.

Reply:
I can clarify that in the text, however. And a link to the above wouldn't hurt.

Reply:
@cobraz: I also just realised it's possible the action could automatically set the GITHUB_TOKEN variable in the process before invoking pulumi, if users don't set it themselves. Actions should always be able to access the `github.token` context. I've not looked into this, however.

Reply:
@jbrunton 
Will this approach work for GitHub Enterprise?

Reply:
@Moon1706 : I don't use GitHub Enterprise so I can't verify it myself, but I don't see any reason why not: `${{ secrets.GITHUB_TOKEN }}` should always be set, even for Enterprise repos. (This is explicitly mentioned in the [docs](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#how-the-permissions-are-calculated-for-a-workflow-job).)

Reply:
Maybe also add this link about GITHUB_TOKEN and rate limits as it mentions GH Enterprise https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#requests-from-github-actions

Reply:
@1oglop1: Thanks for the suggestion. I updated the PR with that link.

Reply:
We already set the Github-token, it's used for pull request comments. We can easily reuse that.

Reply:
@cobraz: Something like [this](https://github.com/pulumi/actions/compare/master...jbrunton:pulumi-actions:use-github-token) instead?

I'd be happy to contribute a PR, but would appreciate a design review first. I couldn't see a better way to pass the token to Pulumi but the above change works.